javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -194,8 +194,15 @@ object BigDecimal {
     else 0
   }
 
-  private[math] def newArrayOfPows(len: Int, pow: Int): Array[Long] =
-    new Array[Long](len - 1).scanLeft[Long, Array[Long]](1)((z, e) => z * pow)
+  private[math] def newArrayOfPows(len: Int, pow: Int): Array[Long] = {
+    val result = new Array[Long](len)
+    var nextValue = 1L
+    for (i <- 0 until len) {
+      result(i) = nextValue
+      nextValue *= pow
+    }
+    result
```

Comment:
Why not just `Array.iterate(1L, len)(_ * pow)`? Shouldn't this expand into the same anyways past optimizer?